### PR TITLE
[core] Make Map.addImage tests pass

### PR DESCRIPTION
**Pull request by asheemmamoowala**
Fix Node tests suite implementation to parse pixelRatio from style json
Premultiply images in the node binding before sending to mbgl core

### CODE REVIEW

File: platform/node/src/node_map.cpp [hunk 1]
```diff
@@ -682,6 +682,19 @@ void NodeMap::AddImage(const Nan::FunctionCallbackInfo<v8::Value>& info) {
...
+    //Pre-multiply the image alpha channel
+    for (size_t i =0 ; i < imageLength ;i+=4) {
+        uint8_t& r = imageData[i + 0];
+        uint8_t& g = imageData[i + 1];
+        uint8_t& b = imageData[i + 2];
+        uint8_t& a = imageData[i + 3];
+        r = (r * a + 127) / 255;
+        g = (g * a + 127) / 255;
+        b = (b * a + 127) / 255;
+    }
```
Comment (kkaefer) [inline platform/node/src/node_map.cpp]: We already have this code in `src/mbgl/util/premultiply.hpp`. Instead, can we move that header to the public include directory, and create an, `UnassociatedImage`, then call `util::premultiply` on it?

File: include/mbgl/util/image.hpp [hunk 2]
```diff
@@ -40,11 +40,16 @@ class Image : private util::noncopyable {
 
     Image(Image&& o)
         : size(o.size),
-          data(std::move(o.data)) {}
+          data(std::move(o.data)) {
+        o.data = nullptr;
```
Comment (jfirebaugh) [inline include/mbgl/util/image.hpp]: This is done automatically when the `std::unique_ptr` is moved, right?

Comment (asheemmamoowala) [inline include/mbgl/util/image.hpp]: Ahh, that is a good point, should've checked the docs on that.

**Comment by 1ec5** [comment]
The submodule pin went stale after mapbox/mapbox-gl-js#4654 got rebased on merge. (If a GL JS commit we pin to gets rebased or squashed, we need to update the pin here afterwards; otherwise new contributors become unable to clone the repository from scratch.) Fixed in #8894.

**Pull request merged**